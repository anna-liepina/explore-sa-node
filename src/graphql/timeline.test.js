describe('GraphQL: Timeline', () => {
    describe('timelineSearch [presets: perPage 3]', () => {
        it(`without criteria`, async () => {
            const { data: { timelineSearch: data } } = await query({
                query: `
                {
                    timelineSearch(perPage: 3) {
                        date
                        avg
                        count
                        postcode
                    }

                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`without criteria: 2nd page`, async () => {
            const { data: { timelineSearch: data } } = await query({
                query: `
                {
                    timelineSearch(perPage: 3, page: 2) {
                        date
                        avg
                        count
                        postcode
                    }

                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`in date range [::from, ::to]`, async () => {
            const { data: { timelineSearch: data } } = await query({
                query: `
                {
                    timelineSearch(from: "2010-01-01", to: "2021-01-01", perPage: 3) {
                        date
                        avg
                        count
                        postcode
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`by postcode wildcard [::pattern]`, async () => {
            const { data: { timelineSearch: data } } = await query({
                query: `
                {
                    timelineSearch(pattern: "E20 1A", perPage: 3) {
                        date
                        avg
                        count
                        postcode
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`by exact postcodes [::postcodes]`, async () => {
            const { data: { timelineSearch: data } } = await query({
                query: `
                {
                    timelineSearch(postcodes: ["E20"], perPage: 3) {
                        date
                        avg
                        count
                        postcode
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });
});
