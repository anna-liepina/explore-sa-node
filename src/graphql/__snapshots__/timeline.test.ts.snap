// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL: Timeline timelineSearch [presets: perPage 3] by exact postcodes [::postcodes] 1`] = `
[
  {
    "avg": 229400,
    "count": 1,
    "date": "2007-08-01",
    "postcode": "E20",
  },
  {
    "avg": 100000,
    "count": 1,
    "date": "2013-10-01",
    "postcode": "E20",
  },
  {
    "avg": 127500,
    "count": 2,
    "date": "2013-12-01",
    "postcode": "E20",
  },
]
`;

exports[`GraphQL: Timeline timelineSearch [presets: perPage 3] by postcode wildcard [::pattern] 1`] = `
[
  {
    "avg": 229400,
    "count": 1,
    "date": "2007-08-01",
    "postcode": "E20",
  },
  {
    "avg": 100000,
    "count": 1,
    "date": "2013-10-01",
    "postcode": "E20",
  },
  {
    "avg": 127500,
    "count": 2,
    "date": "2013-12-01",
    "postcode": "E20",
  },
]
`;

exports[`GraphQL: Timeline timelineSearch [presets: perPage 3] in date range [::from, ::to] 1`] = `
[
  {
    "avg": 100000,
    "count": 1,
    "date": "2013-10-01",
    "postcode": "E20",
  },
  {
    "avg": 127500,
    "count": 2,
    "date": "2013-12-01",
    "postcode": "E20",
  },
  {
    "avg": 755685,
    "count": 3,
    "date": "2015-10-01",
    "postcode": "E20",
  },
]
`;

exports[`GraphQL: Timeline timelineSearch [presets: perPage 3] without criteria 1`] = `
[
  {
    "avg": 229400,
    "count": 1,
    "date": "2007-08-01",
    "postcode": "E20",
  },
  {
    "avg": 100000,
    "count": 1,
    "date": "2013-10-01",
    "postcode": "E20",
  },
  {
    "avg": 127500,
    "count": 2,
    "date": "2013-12-01",
    "postcode": "E20",
  },
]
`;

exports[`GraphQL: Timeline timelineSearch [presets: perPage 3] without criteria: 2nd page 1`] = `
[
  {
    "avg": 755685,
    "count": 3,
    "date": "2015-10-01",
    "postcode": "E20",
  },
  {
    "avg": 745000,
    "count": 1,
    "date": "2015-11-01",
    "postcode": "E20",
  },
  {
    "avg": 715000,
    "count": 2,
    "date": "2015-12-01",
    "postcode": "E20",
  },
]
`;
